### PR TITLE
fix #266 - helpful error message when input address is not in intentions list

### LINKED ISSUE #266: Cannot nominate any non-intention address
Using polkadot-js-ui v0.19.51

I have an account that wants to nominate :)

I can pretty much nominate any address (I tested a few):
image
If you need to copy/paste , this is the address: `5END82tfD39fvgwxe9qCkZJxtyQCtFJXzeSnBXpAR2D7vhZT `

The next however, does not work!
image
The address is `5HMeowwuJACuZwRLFbNMxgdKKKKGKjiFVUFrPF2tayK5e3rT `

So @Tbaut I am sorry, I cannot nominate you!

The addresses have the same length:
```
OK       5END82tfD39fvgwxe9qCkZJxtyQCtFJXzeSnBXpAR2D7vhZT
NOT OK   5HMeowwuJACuZwRLFbNMxgdKKKKGKjiFVUFrPF2tayK5e3rT
```

### PR DESCRIPTION
**Pull request by yjkimjunior**
#266

### CODE REVIEW

File: packages/app-staking/src/StakeList/Nominating.tsx [hunk 1]
```diff
@@ -104,6 +106,20 @@ class Nominating extends React.PureComponent<Props> {
...
+          isNomineeValid
+          ?
+          null
+          :
+          <p> The address you input is not intending to stake, and is therefore invalid. Please try again with a different address.</p>
+        }
+        {
+          isAddressFormatValid
+          ?
+          null
```
[the remark this answers is not in the document]
Comment (yjkimjunior) [reply packages/app-staking/src/StakeList/Nominating.tsx]: not so sure about this one. made some changes here but not so sure how much more readable this is compared to previous

[the remark this answers is not in the document]
Comment (jacogr) [reply packages/app-staking/src/StakeList/Nominating.tsx]: What we do elsewhere -

```
test // line 1
  ? value_one // new line, 2 spaces before
  : value_two // new line, 2 spaces before
```

File: packages/app-staking/src/StakeList/Nominating.tsx [hunk 2]
```diff
@@ -104,15 +108,32 @@ class Nominating extends React.PureComponent<Props> {
...
+        }
       </Modal.Content>
     ];
   }
 
   private onChangeNominee = (nominee: string) => {
     const { intentions } = this.props;
 
+    const publicKey = addressDecode(nominee);
```
Comment (jacogr) [inline packages/app-staking/src/StakeList/Nominating.tsx]: Great. Small issue remains, the function throws, so just wrap in try/catch, see decoding at eg. https://github.com/polkadot-js/common/blob/master/packages/util-keyring/src/address/decode.ts#L16

File: packages/app-staking/src/StakeList/Nominating.tsx [hunk 3]
```diff
@@ -104,17 +109,55 @@ class Nominating extends React.PureComponent<Props> {
...
-      isNomineeValid: intentions.includes(nominee),
-      nominee
-    });
+    try {
+      this.setState({
+        isNomineeValid: intentions.includes(nominee),
+        nominee
+      });
+
+      const publicKey = addressDecode(nominee);
```
Comment (jacogr) [inline packages/app-staking/src/StakeList/Nominating.tsx]: I would just try/catch around this, since the underlying function will throw.

File: packages/app-staking/src/StakeList/Nominating.tsx [hunk 4]
```diff
@@ -104,17 +109,55 @@ class Nominating extends React.PureComponent<Props> {
           onChange={this.onChangeNominee}
           value={nominee}
         />
+        {
+          !isNomineeValid
+          ? <Labelled>
```
Comment (jacogr) [inline packages/app-staking/src/StakeList/Nominating.tsx]: Not entirely excited about the use of this since we basically ignore what it is for however, we don't have a box-y component, so you made do with what you had. No issues there.

(We probably need something in ui-app that does a "display a box with border", e.g. if you look at Transfer there is something that displays the fees, but completely local to there).

TL;DR - Not excited, but it is ok since there is little alternative atm

Comment (yjkimjunior) [reply packages/app-staking/src/StakeList/Nominating.tsx]: We could add an optional prop to the Labelled component for the semantic-react-ui's Label props of color and pointing: https://react.semantic-ui.com/elements/label/#types-pointing-colored

Comment (jacogr) [reply packages/app-staking/src/StakeList/Nominating.tsx]: Ahh, yes, I actually pointed Luke to that a while back. We should obviously not over-use it, but I think it makes perfect sense.

Maybe not use Labelled for that since it is actually different from the SUI "Label", but rather call it something else and just wrap that properly. (Not sure what on the naming). In Labelled we actually fight against the SUI Label, the intent is different.

File: packages/ui-app/src/InputError.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+import React from 'react';
+import Label from 'semantic-ui-react/dist/commonjs/elements/Label/Label';
+
+import classes from './util/classes';
+
+type Props = BareProps & {
+  label?: any
+};
+
+const defaultLabel: any = (// node?
```
Comment (jacogr) [inline packages/ui-app/src/InputError.tsx]: Yes, `React.ReactNode`

File: packages/ui-app/src/InputError.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+import { BareProps } from './types';
+
+import React from 'react';
+import Label from 'semantic-ui-react/dist/commonjs/elements/Label/Label';
+
+import classes from './util/classes';
+
+type Props = BareProps & {
+  label?: any
```
Comment (jacogr) [inline packages/ui-app/src/InputError.tsx]: `React.ReactNode` (tyr to steer clear from any as much as possible - yes, it does happen that they are needed)
